Document freezing of stylistic rules (#5829)
stylelint/stylelint: docs/user-guide/rules/about.md

For example:

```
"shorthand-property-no-redundant-values"
 ↑                  ↑
 the thing          what the rule is checking
```

There is no first part when the rule applies to the whole stylesheet.

For example:

```
"no-duplicate-selectors"
"max-nesting-depth"
 ↑
 what the rules are checking
```

_Rules are named to encourage explicit, rather than implicit, options._ For example, `font-weight-notation: "numeric"|"named-where-possible"` rather than `font-weight-numeric: "always"|"never"`. As `font-weight-named: "never"` _implies_ always numeric, whereas `font-weight-notation: "numeric"` makes it _explicit_.

### No rules

Most rules require _or_ disallow something.

> For example, whether numbers _must_ or _must not_ have a leading zero:

Reviewer: [suggest] It seems necessary to change also this sentence: ⬇️ 

```suggestion
For example, whether colors _must_ or _must not_ be named:
```

- `color-named`: `string - "always-where-possible"|"never"`

>   - `"always"` - there _must always (where possible)_ be named colors
>   - `"never"` - there _must never_ be named colors

Reviewer: [suggest] It seems better to keep consistency with the sentence above ("For example, whether ..."):

```suggestion
  - `"always-where-possible"` - colors _must always (where possible)_ be named
  - `"never"` - colors _must never_ be named
```
